src/Pirouette/SMT/FromTerm.hs

```diff
@@ -161,10 +161,13 @@ constructorFromPIR ::
   [typeVariable] ->
   (Name, TypeMeta builtins meta) ->
   TranslatorT m (String, [(String, PureSMT.SExpr)])
-constructorFromPIR tyVars (name, constructorType) = do
+constructorFromPIR tyVars (name, constructorType) = handleError $ do
   -- Fields of product types must be named: we append ids to the constructor name
   let fieldNames = map (((toSmtName name ++ "_") ++) . show) [1 :: Int ..]
       (_, unwrapped) = Raw.tyUnwrapBinders (length tyVars) constructorType
       (args, _) = Raw.tyFunArgs unwrapped
   cstrs <- zip fieldNames <$> mapM translateType args
   return (toSmtName name, cstrs)
+  where
+    handleError :: TranslatorT m a -> TranslatorT m a
+    handleError m = m `catchError` (throwError . (("At " ++ show name ++ ":") ++))
```

Comment:
Also pointing to the function where this happened (`constructorFromPIR`) might be useful in the future.

Reply:
Also, is this effectively [`withError`](https://hackage.haskell.org/package/mtl-2.3/docs/Control-Monad-Error-Class.html#v:withError)?

Reply:
We get the info that this is from `constructorFromPIR` from the `HasCallStack`. Indeed, this is `withError`, will change!

Reply:
hmm... we're on an older `mtl` and it doesn't have `withError` easily reachable.